databricks_cli/unity_catalog/delta_sharing_cli.py

```diff
@@ -0,0 +1,484 @@
+        click.echo(mc_pretty_format(updated_provider))
+    else:
+        json_cli_base(recipient_profile_json_file, recipient_profile_json,
+                      lambda json: UnityCatalogApi(api_client).update_provider(name, new_name,
+                                                                               comment, json))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='List shares of a provider.')
+@click.option('--name', required=True,
+              help='Name of the provider.')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def list_provider_shares_cli(api_client, name):
+    """
+    List a provider's shares.
+
+    Calls the 'listProviderShares' RPC endpoint of the Unity Catalog service.
+    Returns array of ProviderShare.
+
+    """
+    shares_json = UnityCatalogApi(api_client).list_provider_shares(name)
+    click.echo(mc_pretty_format(shares_json))
+
+
+@click.command(context_settings=CONTEXT_SETTINGS,
+               short_help='Delete a provider.')
+@click.option('--name', required=True,
+              help='Name of the provider to delete.')
+@debug_option
+@profile_option
+@eat_exceptions
+@provide_api_client
+def delete_provider_cli(api_client, name):
+    """
+    Delete a provider.
+
+    Calls the 'deleteProvider' RPC endpoint of the Unity Catalog service.
+    Returns nothing.
+
+    """
+    UnityCatalogApi(api_client).delete_provider(name)
+
+
+def register_delta_sharing_commands(cmd_group):
+    # Share cmds:
+    cmd_group.add_command(create_share_cli, name='create-share')
+    cmd_group.add_command(list_shares_cli, name='list-shares')
+    cmd_group.add_command(get_share_cli, name='get-share')
+    cmd_group.add_command(update_share_cli, name='update-share')
+    cmd_group.add_command(delete_share_cli, name='delete-share')
+    cmd_group.add_command(list_share_permissions_cli, name='list-share-permissions')
+    cmd_group.add_command(update_share_permissions_cli, name='update-share-permissions')
+
+    # Recipient cmds:
+    cmd_group.add_command(create_recipient_cli, name='create-recipient')
+    cmd_group.add_command(list_recipients_cli, name='list-recipients')
+    cmd_group.add_command(get_recipient_cli, name='get-recipient')
+    cmd_group.add_command(rotate_recipient_token_cli, name='rotate-recipient-token')
+    cmd_group.add_command(list_recipient_permissions_cli, name='list-recipient-permissions')
+    cmd_group.add_command(delete_recipient_cli, name='delete-recipient')
+
+    # Provider cmds:
+    cmd_group.add_command(create_provider_cli, name='create-provider')
+    cmd_group.add_command(list_providers_cli, name='list-providers')
+    cmd_group.add_command(get_provider_cli, name='get-provider')
+    cmd_group.add_command(update_provider_cli, name='update-provider')
+    cmd_group.add_command(delete_provider_cli, name='delete-provider')
+    cmd_group.add_command(list_provider_shares_cli, name='list-provider-shares')
```

Comment:
probably we should introduce a helper method to avoid repetition:

```python
util.command_group('share', {
  'create': create_share_cli,
  'list': list_shares_cli,
  'get': get_share_cli,
  'update': update_share_cli,
  'delete': delete_share_cli,
  'permissions': list_share_permissions_cli,
  'permissions-update': update_share_permissions_cli, # TBD: maybe there's a better way
})

util.command_group('share-recipient', {
  ..
})

util.command_group('share-provider', {
  ..
})
```

Reply:
for delta sharing permissions, commands should go into permissions CLI. we currently don't have a consistent permissions experience in cli - e.g. `databricks cluster permissions get abcd124-1252-3522`, so better keep the change surface to minimum, until permissions semantics are properly introduced in CLI. 

`click` library has limitations on command name chaining and all naming options would be reviewed in the future designs.

Reply:
Should pull in the Delta Sharing folks on this question (Xiaotong Sun, William Chau, etc.)

Reply:
Consolidation can happen at a later point.